apps/src/javalab/JavalabEditor.jsx

```diff
@@ -90,7 +90,10 @@ class JavalabEditor extends React.Component {
     e.preventDefault();
     const {filename, setFilename} = this.props;
     const {newFilename} = this.state;
-    renameProjectFile(filename, newFilename);
+    // We don't want to actually save the file if we're editing startSources.
+    if (!window.appOptions.level.editBlocks) {
```

Comment:
This could just be that our code base is odd, but it seems odd that we'd be getting this value from the window object. Is there another way to get it here?

Reply:
I could pass it in to JavalabView -> JavalabEditor as a prop from Javalab.js, would that be better?

Reply:
with the changes in [this PR](https://github.com/code-dot-org/code-dot-org/pull/40049/files#diff-13f5c29bba076ae535f1304a363c8a0e63661c75b07f91b89d38d633a8aa3aa0R96) this code won't be necessary anymore, since all changes are going through our project save system (either manually via commit code or via autosaves)